Modules/FindCUDA/select_compute_arch.cmake

```diff
@@ -0,0 +1,176 @@
+# Borrowed from caffe
+# https://github.com/BVLC/caffe/blob/master/cmake/Cuda.cmake
+
+# Synopsis:
+# Function for selecting GPU arch flags for nvcc based on CUDA_ARCH_NAME
+# Usage:
+#   SELECT_NVCC_ARCH_FLAGS(out_variable)
+#
+# Variables affecting the choice:
+#
+# CUDA_ARCH_NAME: One of ("Fermi" "Kepler" "Kepler-M" "Maxwell" "Maxwell-M" "All" "Manual")
+# ENV{CUDA_ARCH_BIN} : set if CUDA_ARCH_NAME set to "Manual"
+#
+
+# Known NVIDIA GPU achitectures
+# This list will be used for CUDA_ARCH_NAME = All option
+
+if (NOT CUDA_arch_select_included)
```

Comment:
I found the inclusion of this file via cuda_find_helper_file from FindCUDA makes names like DETECT_INSTALLED_GPUS be hidden from the outside callers, which broke some existing Torch code where I extracted FindCudaArch.cmake from. So for backward compatibility I have provided FindCudaArch.cmake in my repo that includes this file via include(). Since I was not sure what happens if I include it twice from 2 different places I have added those. Probably should be removed, what woudl you say ?